libs/components/filter-bar/testing/src/modules/filter-bar/filter-bar-harness.ts

```diff
@@ -0,0 +1,106 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+import { SkySelectionModalHarness } from '@skyux/lookup/testing';
+
+import { SkyFilterBarHarnessFilters } from './filter-bar-harness-filters';
+import { SkyFilterBarItemHarness } from './filter-bar-item-harness';
+import { SkyFilterBarItemHarnessFilters } from './filter-bar-item-harness-filters';
+
+/**
+ * Harness for interacting with a filter bar component in tests.
+ */
+export class SkyFilterBarHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-filter-bar';
+
+  #documentRootLocator = this.documentRootLocatorFactory();
+
+  #getClearFiltersButton = this.locatorForOptional(
+    '.sky-filter-bar-clear-filters',
+  );
+  #getFilterPickerButton = this.locatorForOptional(
+    '.sky-filter-bar-filter-picker',
+  );
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyFilterBarHarness` that meets certain criteria.
+   */
+  public static with(
+    filters: SkyFilterBarHarnessFilters,
+  ): HarnessPredicate<SkyFilterBarHarness> {
+    return SkyFilterBarHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Clicks the clear filters button.
+   */
+  public async clickClearFilters(): Promise<void> {
+    const button = await this.#getClearFiltersButton();
+    if (!button) {
+      throw new Error('Clear filters button not found');
```

Comment:
Since the logic for if the clear button shows or not is because of our internal logic, can we give consumers a better idea of why the clear filters button is not found

Reply:
added